src/common/hooks/useInjectCompanyChanges.ts

```diff
@@ -35,7 +37,9 @@ export function useInjectCompanyChanges(
       return;
     }
 
-    dispatch(injectInChanges({ object: 'company', data: company }));
+    if (isCompanySettingsActive) {
+      dispatch(injectInChanges({ object: 'company', data: company }));
+    }
```

Comment:
I am no sure this works? I believe we inject company settings _outisde_ settings, this means that case won't work unless we are in /settings part?

Reply:
@beganovich Hmm, you are probably right about one part of logic. I think this part was not present in the `useInjectCompanyChanges` when I began working on `group_settings` in June:

`if (companyChanges && options?.overwrite === false)`

What's happening here is that when we click on `configureGroupSettings`, `companyChanges` will be populated, and the `first part of the if statement will be true`. However, the second part is true in only one place. So, we only have `overwrite` set to false in `options` at that one location. If we change the statement to use the `|| operator`, perhaps the logic will work without needing to check. 

`if (isCompanySettingsActive)`

Without that change, when we trigger `configureGroupSettings`, the changes will be set to the `group_settings object`, but simultaneously, wherever the `useInjectCompanyChanges` is called, `company_changes` will revert to the `company level`.

I hope you understood why the if statement is necessary. If not, please let me know, and I'm happy to explain it one more time.

Let me know your thoughts.